Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -1327,7 +1327,11 @@ private FileOrDirectoryArtifact GetDeclaredArtifact(FileArtifact file)
             return file;
         }
 
-        private async Task<Possible<Unit>> TryHashFileArtifactsAsync(PipArtifactsState state, OperationContext operationContext, bool allowUndeclaredSourceReads)
+        private async Task<Possible<Unit>> TryHashFileArtifactsAsync(
+            PipArtifactsState state, 
+            OperationContext operationContext, 
+            bool allowUndeclaredSourceReads, 
+            string pipDescription = null)

```

Comment:
Do you need to pass `pipDescription` explicitly here?  There is `state.PipInfo.Description` which seems to be exactly what you want.

Reply:
Now I see that `state.PipInfo` might not even be set at this point (mutable fields...)

Reply:
Yes, I tried it out first, but it was null in most cases